src/averageAges.js

```diff
@@ -55,10 +82,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const validPeople = people.filter(person => {
+    if (onlyWithSon) {
+      return person.sex === 'm' && person.mother !== undefined;
+    } else {
+      return person.mother !== undefined;
+    }
+  });
+
+  const { totalAgeDiff, count } = validPeople.reduce((accumulator, pair) => {
+    const child = pair;
+    const mother = people.find(person => person.name === pair.mother);
+
+    if (child && mother) {
+      const ageDiff = child.born - mother.born;
+
+      return {
+        totalAgeDiff: accumulator.totalAgeDiff + ageDiff,
+        count: accumulator.count + 1,
+      };
+    } else {
+      return accumulator;
+    }
+  }, {
+    totalAgeDiff: 0, count: 0,
+  });
+
+  const averageAgeDiff = count > 0 ? totalAgeDiff / count : 0;
+
+  return parseFloat(averageAgeDiff.toFixed(2));
```

Comment:
You are converting the 'averageAgeDiff' to a string using the .toFixed() method and then converting it back to a number using parseFloat(). This is unnecessary. You can use the .toFixed() method if you want to display the number, but you need to keep it as a number for internal calculations.